starport/services/networkbuilder/simulation.go

```diff
@@ -0,0 +1,219 @@
+package networkbuilder
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"context"
+	"github.com/cenkalti/backoff"
+	"github.com/otiai10/copy"
+	"github.com/pelletier/go-toml"
+	"github.com/tendermint/starport/starport/pkg/availableport"
+	"github.com/tendermint/starport/starport/pkg/cmdrunner"
+	"github.com/tendermint/starport/starport/pkg/cmdrunner/step"
+	"github.com/tendermint/starport/starport/pkg/cosmosver"
+	"github.com/tendermint/starport/starport/pkg/gomodulepath"
+	"github.com/tendermint/starport/starport/pkg/httpstatuschecker"
+	"github.com/tendermint/starport/starport/pkg/xurl"
+	"github.com/tendermint/starport/starport/services/chain"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const ErrValidatorSetNil = "validator set is nil in genesis and still empty after InitChain"
+
+// VerifyProposals generates a genesis file from the current launch information and proposals to verify
+// The function returns false if the generated genesis is invalid
+func (b *Builder) VerifyProposals(ctx context.Context, chainID string, proposals []int, commandOut io.Writer) (bool, error) {
```

Comment:
Let's use internal event system to send messages to listeners (in this case `cmd`) instead of a `commandOut`. Similar to this one: https://github.com/tendermint/starport/blob/49aa840/starport/services/networkbuilder/networkbuilder.go#L88

Reply:
The output is not very clean using the event bus. I think this is not adapted because this is the whole execution output of a command while events should just print a description of what is going on. Maybe we could have a specific package to handle this kind of output, for now I think it's better to pass the stdout from the command line. (`--debug` is not supposed to be used by the end user anyway)